macrosynergy/dataquery/api.py

```diff
@@ -147,25 +149,21 @@ def _fetch_threading(self, endpoint, params: dict, server_count: int = 5):
         # method chosen.
         url = self.access.base_url + endpoint
         select = "instruments"
-
+        response = {}
         results = []
         counter = 0
-        while counter <= server_count:
+        while (not (select in response.keys())) and (counter <= server_count):
             try:
                 # The required fields will already be instantiated on the instance of the
                 # Class.
-                response: dict = self.access.get_dq_api_result(url=url, params=params, proxy=self.proxy)
+                response, status, msg  = self.access.get_dq_api_result(url=url, params=params, 
+                                                                    proxy=self.proxy)
             except ConnectionResetError:
                 counter += 1
                 time.sleep(0.05)
                 print(f"Server error: will retry. Attempt number: {counter}.")
```

Comment:
Changed print statement to log statement. 
Breaking here would imply that the error tickers are not retried by the thread driver (`_request`). 
The thread driver breaks the code in case of an erroneous result, and re-runs the `error_tickers`. 
My concern with retrying individual threads is that it could result in too many hits to dataquery